### PR TITLE
chore(deps): Bump logging-interceptor to 4.0.0 and fix errors

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #2018

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/di/Modules.kt [hunk 1]
```diff
@@ -259,7 +259,7 @@ val networkModule = module {
         val builder = OkHttpClient().newBuilder()
             .connectTimeout(connectTimeout.toLong(), TimeUnit.SECONDS)
             .readTimeout(readTimeout.toLong(), TimeUnit.SECONDS)
-            .addInterceptor(get())
+            .addInterceptor(RequestAuthenticator(get()) as Interceptor)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/di/Modules.kt]: Casting is bad

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/di/Modules.kt]: Sorry, It is not required now as the response is not nullable now. Updated.

File: app/src/main/java/org/fossasia/openevent/general/di/Modules.kt [hunk 2]
```diff
@@ -259,7 +258,7 @@ val networkModule = module {
         val builder = OkHttpClient().newBuilder()
             .connectTimeout(connectTimeout.toLong(), TimeUnit.SECONDS)
             .readTimeout(readTimeout.toLong(), TimeUnit.SECONDS)
-            .addInterceptor(get())
+            .addInterceptor(RequestAuthenticator(get()))
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/di/Modules.kt]: Why not just get()?

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/di/Modules.kt]: Displaying error-
```
@JvmName public final inline fun addInterceptor(crossinline block: (chain: Interceptor.Chain) -> Response): OkHttpClient.Builder defined in okhttp3.OkHttpClient.Builder
public final fun addInterceptor(interceptor: Interceptor): OkHttpClient.Builder defined in okhttp3.OkHttpClient.Builder
```

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/di/Modules.kt]: You are passing `RequestAuthenticator(get())` and `single { RequestAuthenticator(get()) }` is already defined above.  Either remove that or use get

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/di/Modules.kt]: Updated

**Pull request merged**